assets/src/edit-story/app/uploader/useUploader.js

```diff
@@ -97,18 +104,22 @@ function useUploader() {
         throw createError('SizeError', file.name, message);
       }
 
-      if (!isValidType(file) && !canTranscodeFile(file)) {
-        /* translators: %s is a list of allowed file extensions. */
-        const message = sprintf(
-          /* translators: %s: list of allowed file types. */
-          __('Please choose only %s to upload.', 'web-stories'),
-          allowedFileTypes.join(
-            /* translators: delimiter used in a list */
-            __(', ', 'web-stories')
-          )
-        );
-
-        throw createError('ValidError', file.name, message);
+      // The file type is not supported by default, but maybe it can be transcoded?
+      if (!isValidType(file)) {
+        // Nope, doesn't look like it! Bail now because it cannot be uploaded.
+        if (!isTranscodingEnabled(file)) {
```

Comment:
```suggestion
      if (!isValidType(file) && !isTranscodingEnabled(file)) {
```
Is it cleaner to have have one if statement?

Reply:
I split it up to make it easier to comprehend with the inline documentation. No strong preference though.

Reply:
At some point should this error messaging look like this?

```
if ( isVideo( file ) && isTranscodingEnabled(file) && !canTranscodeFileType(file)) {
    ... "Unable to transcode video as unsupported file type

if (!isValidType(file)) {
   ... "Please choose only %s to upload."
```

I guess that will be happened in https://github.com/google/web-stories-wp/issues/6053

Reply:
Yes. These error messages right now are far from final.